packages/lit-dev-tools-esm/src/fake-github-middleware.ts

```diff
@@ -394,4 +401,71 @@ class FakeGitHub {
     this._gists.set(gist.id, gist);
     return jsonResponse(ctx, 200, gist);
   }
+
+  /**
+   * Simulates PATCH https://api.github.com/gists/<id>, the API for creating a
+   * new revision of an existing GitHub gist.
+   *
+   * Documentation:
+   * https://docs.github.com/en/rest/reference/gists#update-a-gist
+   */
+  async updateGist(ctx: Koa.Context) {
+    ctx.set('Access-Control-Allow-Origin', '*');
+    const accept = ctx.get('accept');
+    if (accept !== 'application/vnd.github.v3+json') {
+      return jsonResponse(ctx, 415, {
+        message: "Unsupported 'Accept' header",
+      });
+    }
+
+    const authMatch = (ctx.get('authorization') ?? '').match(
+      /^\s*(?:token|bearer)\s(?<token>.+)$/
+    );
+    const authToken = authMatch?.groups?.token?.trim() ?? '';
+    const userAndScope = this._accessTokens.get(authToken);
+    if (!userAndScope) {
+      return jsonResponse(ctx, 401, {message: 'Bad credentials'});
+    }
+
+    const idMatch = ctx.path.match(/^\/gists\/(?<id>.+)/);
+    const id = idMatch?.groups?.id;
+    const gist = this._gists.get(id ?? '');
+    if (!gist) {
+      // TODO(aomarks) Check error code
+      return jsonResponse(ctx, 404, {message: 'Invalid gist'});
+    }
+
+    let updateRequest: UpdateGistRequest;
+    try {
+      updateRequest = await coBody.json(ctx);
+    } catch (e) {
+      return jsonResponse(ctx, 400, {message: 'Problems parsing JSON'});
+    }
+    if (!updateRequest.files || Object.keys(updateRequest.files).length === 0) {
+      return jsonResponse(ctx, 422, {message: 'Invalid files'});
+    }
+
+    // TODO(aomarks) In the real API, each revision is stored individually, and
+    // you can GET either the latest revision, or a specific revision. Until
+    // needed, though, this fake just directly updates the simplified flat gist
+    // instead.
+    const newFiles = updateRequest.files;
+    for (const [newFilename, newFile] of Object.entries(newFiles)) {
+      newFile.filename = newFilename;
+    }
+    const oldFiles = gist.files;
+    for (const [oldFilename, oldFile] of Object.entries(oldFiles)) {
+      // To delete a file, you set its content to an empty string. If omitted,
+      // the existing file is unchanged. See
+      // https://github.community/t/deleting-or-renaming-files-in-a-multi-file-gist-using-github-api/170967
+      if (newFiles[oldFilename] === undefined) {
+        newFiles[oldFilename] = oldFile;
+      } else if (!newFiles[oldFilename]?.content) {
```

Comment:
Worth making this more specific to ensure there is a `content` prop set and it's exactly `""`?

Reply:
Content can be empty string or undefined to trigger a delete. Updated comment to clarify.